client/src/chat/Chat.jsx

```diff
@@ -0,0 +1,178 @@
+import React, { useEffect, useRef, useState } from "react";
+import { usePlayer, useStageTimer, usePlayers } from "@empirica/core/player/classic/react";
+import { Loading } from "@empirica/core/player/react";
+import { MessageBubble } from "./MessageBubble";
+import { TextBar } from "./TextBar";
+import { ChatIcon } from "./Icons";
+import { reconstructChatState, getNextActionId } from "./chatUtils";
+
+export function Chat({
+  scope,
+  attribute = "chat",
+  showNickname = true,
+  showTitle = false,
+  reactionEmojisAvailable = [],
+  reactToSelf = true,
+  numReactionsPerMessage = 1,
+}) {
+  const player = usePlayer();
+  const players = usePlayers();
+  const progressLabel = player.get("progressLabel");
+  const stageTimer = useStageTimer();
+  const [messages, setMessages] = useState([]);
+  const scroller = useRef();
+
+  const currentPlayerPosition = player?.get("position");
+
+  // Reconstruct chat state from actions whenever they change
+  useEffect(() => {
+    const actions = scope?.getAttribute(attribute)?.items || [];
+    const reconstructed = reconstructChatState(actions);
+    setMessages(reconstructed);
+  }, [scope?.getAttribute(attribute)?.items?.length, scope, attribute]); // Depend on items length
+
+  // Auto-scroll to bottom when messages change
+  useEffect(() => {
+    if (scroller.current) {
+      scroller.current.scrollTo({
+        top: scroller.current.scrollHeight,
+        behavior: "smooth",
+      });
+    }
+  }, [messages]);
+
+  if (!stageTimer) return null;
+  const elapsed = (stageTimer?.elapsed || 0) / 1000;
+
+  if (!scope || !player) {
+    return <Loading />;
+  }
+
+  const handleSendMessage = (text) => {
+    const actions = scope.getAttribute(attribute)?.items || [];
+    const actionId = getNextActionId(actions);
+
+    scope.append(attribute, {
+      id: actionId,
+      type: "send_message",
+      content: text,
+      targetId: undefined,
+      playerPosition: currentPlayerPosition,
+      sender: {
+        id: player?.id,
+        name: player?.get("name") || player?.id,
```

Comment:
this is the place to make the fallback the based on the player position (e.g. player 1), not falling back to the player id